lib/command/common/run-with-cover.js

```diff
@@ -177,8 +177,10 @@ function run(args, commandName, enableHooks, callback) {
                     hook.unloadRequireCache(matchFn);
                 }
                 // runInThisContext is used by RequireJS [issue #23]
+                // run InNewContext is usefull for mocking deps in unit tests http://howtonode.org/testing-private-state-and-mocking-deps
                 if (config.hooks.hookRunInContext()) {
                     hook.hookRunInThisContext(matchFn, transformer, hookOpts);
+                    hook.hookRunInNewContext(matchFn, transformer, hookOpts);
```

Comment:
This is a potentially breaking change. I would like this to be a separate config option because of that.